File: app/src/main/java/bj/drunkmessengerblocker/blacklist/BlacklistActivity.java
package bj.drunkmessengerblocker.blacklist;

import android.os.Bundle;
import android.support.v7.widget.RecyclerView;
import android.view.MenuItem;

import javax.inject.Inject;

import bj.drunkmessengerblocker.AppComponent;
import bj.drunkmessengerblocker.BaseActivity;
import bj.drunkmessengerblocker.R;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * Created by j on 16/01/2017.
 */

public class BlacklistActivity extends BaseActivity implements BlacklistContract.View
{
    private final String TAG = "BlacklistActivity";

    private BlacklistComponent component;
    @Inject BlacklistPresenter presenter;
    @BindView(R.id.recyclerView) RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_blacklist);
        ButterKnife.bind(this);

        presenter.initialiseRecyclerView(recyclerView);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }


    @Override
    protected void setupComponent(AppComponent appComponent)
    {
        component = DaggerBlacklistComponent.builder()
                .appComponent(appComponent)
                .blacklistModule(new BlacklistModule(this))
                .build();

        component.inject(this);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem menuItem)
    {
        onBackPressed();
        return true;
    }

    @OnClick(R.id.fab)
    public void addName()
    {
        presenter.onFabClicked();
    }
}

File: app/src/main/java/bj/drunkmessengerblocker/blacklist/BlacklistActivity.kt
package bj.drunkmessengerblocker.blacklist

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.view.MenuItem
import bj.drunkmessengerblocker.AppComponent
import bj.drunkmessengerblocker.R
import bj.drunkmessengerblocker.common.BaseActivity
import butterknife.ButterKnife
import butterknife.OnClick
import kotlinx.android.synthetic.main.activity_blacklist.*
import javax.inject.Inject

/**
 * Created by Josh Laird on 23/05/2017.
 */
class BlacklistActivity : BaseActivity(), BlacklistContract.View {
    @Inject lateinit var presenter: BlacklistPresenter
    @Inject lateinit var controller: BlacklistController

    companion object {
        fun createIntent(context: Context): Intent {
            return Intent(context, BlacklistActivity::class.java)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_blacklist)
        ButterKnife.bind(this)

        setupRecyclerView()
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
    }

    override fun setupComponent(appComponent: AppComponent) {
        DaggerBlacklistComponent.builder()
                .appComponent(appComponent)
                .blacklistModule(BlacklistModule(this))
                .build()
                .inject(this)
    }

    override fun onOptionsItemSelected(menuItem: MenuItem): Boolean {
        onBackPressed()
        return true
    }

    @OnClick(R.id.fab)
    fun addName() {
        presenter.onFabClicked()
    }

    private fun setupRecyclerView() {
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.adapter = controller.adapter
        controller.requestModelBuild()
    }
}
